src/transform2d.rs

```diff
@@ -529,6 +530,23 @@ where
             self.transform_point(point2(min.x, max.y)),
         ])
     }
+
+
+    /// Returns a box that encompasses the result of transforming the given box by this
```

Comment:
I believe `transform_rect` and `transform_box` are not sound. They are lossy, and they specifically compute the enclosing bound on the transformed volume. There are many lossy ways to transform rectangles and boxes. For example, another way is to compute the inner bound instead of the external one.

Reply:
I believe they are sound and behave as documented but a more descriptive name would prebably be better? We use transform_rect in WebRender for culling purposes where the enclosing bounds is what we care about.

Maybe `fn outer_transformed_rect/box` would be better ?